Added interfaces

hhan-solvd opened the pull request:
Added interfaces, updated the modifiers of fields and updated a few methods.

hhan-solvd committed:
Updated Customer.java

Implemented interface.

hhan-solvd committed:
Updated CustomerEvaluation.java

Implemented interface and changed fields to final fields.

hhan-solvd committed:
Updated Dealership.java

Updated Dealership.java

hhan-solvd committed:
Updated Employee.java

Implemented interfaces, changed fields to final fields,  and updated addVehicleToInventory(), checkVehicleAvailability() and removeVehicleFromInventory().

hhan-solvd committed:
Updated EvaluationSummary.java

Changed fields to final fields.

hhan-solvd committed:
Updated Finance.java

Implemented interface, changed fields to final fields, added a field called paymentMethod, added getPaymentMethod(), and updated financeOrNot() to determinePaymentMethod().

hhan-solvd committed:
Updated Receipt.java

Changed field to final field.

hhan-solvd committed:
Updated Transaction.java

Changed fields to final fields, deleted the field of paymentType, and updated getPaymentType() and toString().

hhan-solvd committed:
Updated Vehicle.java

Implemented interface.

hhan-solvd committed:
Updated VehicleInventory.java

Implemented interface.

hhan-solvd committed:
Added interfaces

Added interfaces

ankochergin commented on cardealership/Evaluation.java:
```diff
@@ -0,0 +1,9 @@
+package com.solvd.cardealership;
+
+public interface Evaluation {
```
it's recommended to add I letter to interface name, for example IEvaluation

ankochergin commented on cardealership/Customer.java:
```diff
@@ -2,10 +2,10 @@
+public class Customer implements Person{
+    private final String name;
+    private final int phone;
+    private final double budget;
```
why this variables are final? I think customer can change name, phone or/and budget
final should be only constant variables / not changing variables. for example, speed of light is constant
please check all your variables and make sure they should be final or not

ankochergin reviewed (commented):
1. Please add final class
2. Add static method and block as required in HW

hhan-solvd committed:
Updated Customer.java

Added I to the interface's name, changed the modifiers of fields, and added exceptions.

hhan-solvd committed:
Updated CustomerEvaluation.java

Added I to the interface's name, changed the modifiers of fields, changed calculateRating() to a final method, and added exceptions.

hhan-solvd committed:
Updated Dealership.java

Updated System.out.println to logger and added exceptions.

hhan-solvd committed:
Updated ElectricVehicle.java

Added exceptions.

hhan-solvd committed:
Updated Employee.java

Added I to the interfaces’ names, changed the modifiers of fields and added exceptions.

hhan-solvd committed:
Updated IEvaluation.java

Add I to the interface’s name.

hhan-solvd committed:
Updated EvaluationSummary.java

Changed the modifiers of fields, updated this class to a final class and changed System.out.println to logger.

hhan-solvd committed:
Updated Finance.java

Added I to the interface’s name and changed the modifiers of fields.

hhan-solvd committed:
Updated GasVehicle.java

Added exceptions.

hhan-solvd committed:
Updated HybridVehicle.java

Added exceptions.

hhan-solvd committed:
Updated IInventory.java

Added I to the interface’s name.

hhan-solvd committed:
Updated IPaymentMethod.java

Added I to the interface’s name.

hhan-solvd committed:
Updated IPerson.java

Added I to the interface’s name.

hhan-solvd committed:
Updated Receipt.java

Changed the modifier of the field, updated this class to a final class, and changed System.out.println to logger.

hhan-solvd committed:
Updated Transaction.java

Changed the modifiers of the fields.

hhan-solvd committed:
Updated ITransportation.java

Added I to the interface’s name.

hhan-solvd committed:
Updated Vehicle.java

Added I to the interface’s name and added exceptions.

hhan-solvd committed:
Updated VehicleInventory.java

Added I to the interface’s name, static block, static method printVehicleInventory(), and exceptions.

hhan-solvd committed:
Updated IWorker.java

Added I to the interface’s name.

hhan-solvd committed:
Added Exception File

Added exception file

hhan-solvd committed:
Added log4j2.xml

Added configuration file.
